src/server.c

```diff
@@ -6151,25 +6175,81 @@ sds genRedisInfoStringCommandStats(sds info, dict *commands) {
     return info;
 }
 
+void addSectionsToDict(dict * section_dict, char ** sections, int len) {
+    for (int i = 0; i < len; i++){
+        sds section = sdsnew(sections[i]);
+        dictAdd(section_dict, section, NULL);
+    }
+}
+
+
+dict * genSectionDict(client * c, const char * source, int * all_sections, int * everything) {
```

Comment:
some styling fix, and a better function name.
```suggestion
dict *genInfoSectionDict(client *c, const char *source, int *all_sections, int *everything) {
```
please also add a comment describing what this function does, and its input and outputs (especially important when you have output arguments, and it's not clear who's responsible of releasing things).